### ISSUE TITLE
Missing CS8625 for constructor call on static field initialization

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#33394]
**Version Used**: 
16.0.0 Preview 3.0

**Steps to Reproduce**:

```csharp
#nullable enable
public class C
{
    public C(string a) { }
    public static C Empty = new C(null); // Missing CS8625
}
```

**Expected Behavior**:
Calling `C`'s constructor with `null` should generate a warning

**Actual Behavior**:
It doesn't, but does when `static` is removed

**Comment by cston** [comment]
@Tragetaschen, thanks for reporting this issue. Fixed in https://github.com/dotnet/roslyn/pull/33093.
